src/sections/Meshery/Meshery-integrations/IntegrationsGrid.js

```diff
@@ -1,44 +1,173 @@
+        if (item.name === category) item.isSelected = true;
+      });
+    } else {
+      categoryList[0].isSelected = true;
+    }
+
+    setCategoryList(categoryList);
+    setIntegrationCollection();
+
+  };
+
+  const setFilter = (event) => {
+
+    let count = 0;
+    const selectedCategory = event.target.innerHTML;
+
+    if (selectedCategory == "All"){
+      categoryList.forEach(item => {
+        if (item.isSelected & item.name != "All") item.isSelected = false;
+      }
+      );
+
+    }
+
+    categoryList.forEach(item => {
+      if (item.name == selectedCategory) item.isSelected = !item.isSelected;
+      if (item.isSelected && item.name != "All") count++;
+    });
+
+    if (count === 0) categoryList[0].isSelected = true;
+    else categoryList[0].isSelected = false;
+
+    setCategoryList(categoryList);
+    setIntegrationCollection();
+  };
+
+  const setIntegrationCollection = () => {
+
+    if (categoryList[0].isSelected){
+      setIntegrationList(data.allMdx.nodes);
+      return;
+    }
+
+    let tempIntegrationCollection = [];
+
+
+    categoryList.forEach(item => {
+
+      if (item.isSelected){
+
+        data.allMdx.nodes.forEach(integration => {
+          if (integration.frontmatter.category == item.name){
+            tempIntegrationCollection = [...tempIntegrationCollection, integration];
+          }
+        });
+
+      }
+
+    });
+
+    setIntegrationList(tempIntegrationCollection);
+
+  };
+
 
-const IntegrationsGrid = ({ data }) => {
   return (
     <HoneycombGrid>
-      <div className="container">
-        {data.map((item,index) => {
+      <section className="category">
+        {categoryList.map((item) => {
+          return (
+            <p
+              key={item.id}
+              className={item.isSelected ? "items selected" : "items"}
+              onClick={setFilter}
+            >
+              {item.name}
+            </p>
+          );
+        })}
+      </section>
+
+      <ResponsiveHoneycomb
+        size={90}
+        items={IntegrationList}
+        renderItem={(item) => {
+
           const status = item.frontmatter.status === "InProgress" ? true : false;
 
-          const data = status ? (
-            <div style={{ background: "#A0AAAA" }} key={index}>
-              <div style={{ background: "#A0AAAA" }}>
-                <img
-                  src={item.frontmatter.integrationIcon.publicURL}
-                  alt={item.frontmatter.title}
-                  style={{
-                    filter: "brightness(0) invert(1)",
-                  }}
-                />
-              </div>
-            </div>
-          ) : (
-            <div>
-              <Link
-                to={`/service-mesh-management/meshery${item.fields.slug}`}
-              >
-                <div>
+          if (status){
+            return (
+              <Hexagon className="container" style={{ background: "#A0AAAA" }}>
+                <div style={{ background: "#A0AAAA" }}>
```

Comment:
Why do we need this div, again?

Reply:
We need this div to make the border hover effect.

Reply:
Use the div that is rendered by the `HoneyComb` component. No need for an extra div.
And, if that doesn't work, use `::after` to do so.

Reply:
Thank you. Too many layers. The on hover **border** effect should not be made through another div, but... yes, that's right: **borders**.

Reply:
I removed that extra div in the lastest commit.